source/common/config/watch_map.cc

```diff
@@ -54,8 +54,17 @@ absl::flat_hash_set<Watch*> WatchMap::watchesInterestedIn(const std::string& res
 void WatchMap::onConfigUpdate(const Protobuf::RepeatedPtrField<ProtobufWkt::Any>& resources,
                               const std::string& version_info) {
   if (watches_.empty()) {
-    ENVOY_LOG(warn, "WatchMap::onConfigUpdate: there are no watches!");
-    return;
+    if (resources.empty()) {
```

Comment:
Question: For lds/cds type, we always insert a wildcard watcher so that watches_ should never be empty, correct?

Reply:
Correct! (I just took a look at the code to be extra sure, and yes, e.g. in cds_api_impl.h there is `void initialize() override { subscription_->start({}); }`